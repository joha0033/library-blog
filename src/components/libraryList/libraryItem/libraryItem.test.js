import React from 'react';
import ReactDOM from 'react-dom';
import LibraryItem from './libraryItem';
import LibraryList from '../libraryList';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme'
import { mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })

describe('renders a div', () => {
  let props
  let mountedItem
  //mount the list of items
  const item = () => {
    if (!mountedItem) {
      mountedItem = mount(
        <LibraryItem items={mount(<LibraryList />).state().libraries} />
      );
    }
    return mountedItem;
  }

  beforeEach(() => {
    props = {
      form: false
    }
    mountedItem = undefined
  })

  //at least renders 1 div
  it("always renders a div", () => {
    const divs = item().find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

  //at least renders 1 div
  it("always renders a div", () => {
    const divs = item().find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

  describe("the rendered item", () => {

    it("contains everything else that gets rendered", () => {
      const divs = item().find("div");
      const wrappingDiv = divs.first();
      // console.log(item().children().children());
      expect(wrappingDiv.children().length).toEqual(item().children().children().length)
    });

    it("mapping function produces image", () => {
      // const wrapper = mount(<LibraryItem items={mount(<LibraryList />).state().libraries}/>);
      const tree = renderer
        .create(<LibraryItem items={mount(<LibraryList />).state().libraries}/>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
  })

})
