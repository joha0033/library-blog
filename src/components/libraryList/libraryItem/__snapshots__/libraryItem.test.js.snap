// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a div the rendered item mapping function produces image 1`] = `
<div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <img
        alt="containerPic"
        className={undefined}
        src="https://avatars0.githubusercontent.com/u/9921462?s=200&v=4"
      />
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <h1>
          Wallaby.js
        </h1>
      </div>
      <div>
        <p
          className={undefined}
        >
          Lorem ipsum dolor amet sartorial snackwave celiac thundercats, retro plaid marfa tumeric small batch pok pok health goth microdosing cardigan kitsch. Blog occupy blue bottle, put a bird on it VHS seitan live-edge small batch cornhole. Try-hard scenester kombucha schlitz seitan pop-up tumblr sustainable sartorial trust fund master cleanse YOLO four loko organic williamsburg. Offal ugh coloring book kinfolk man braid.
        </p>
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <img
        alt="containerPic"
        className={undefined}
        src="https://blobscdn.gitbook.com/v0/b/gitbook-28427.appspot.com/o/spaces%2F-L5K1I1WsuQMZ8ecEuWg%2Favatar.png?generation=1518623866348435&alt=media"
      />
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <h1>
          Redux
        </h1>
      </div>
      <div>
        <p
          className={undefined}
        >
          Lorem ipsum dolor amet sartorial snackwave celiac thundercats, retro plaid marfa tumeric small batch pok pok health goth microdosing cardigan kitsch. Blog occupy blue bottle, put a bird on it VHS seitan live-edge small batch cornhole. Try-hard scenester kombucha schlitz seitan pop-up tumblr sustainable sartorial trust fund master cleanse YOLO four loko organic williamsburg.
        </p>
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <img
        alt="containerPic"
        className={undefined}
        src="https://www.sydjs.com/images/showbag/moment.jpg"
      />
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <h1>
          Moment
        </h1>
      </div>
      <div>
        <p
          className={undefined}
        >
          Retro plaid marfa tumeric small batch pok pok health goth microdosing cardigan kitsch. Blog occupy blue bottle, put a bird on it VHS seitan live-edge small batch cornhole. Try-hard scenester kombucha schlitz seitan pop-up tumblr sustainable sartorial trust fund master cleanse YOLO four loko organic williamsburg. Offal ugh coloring book kinfolk man braid.
        </p>
      </div>
    </div>
  </div>
</div>
`;
